File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/model/security/BooksController.kt
package io.micronaut.views.model.security

import io.micronaut.context.annotation.Requires
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.security.annotation.Secured
import io.micronaut.security.rules.SecurityRule
import io.micronaut.views.View

@Requires(property = "spec.name", value = "SecurityViewModelProcessorSpec")
//tag::class[]
@Controller("/")
class BooksController {

    @Secured(SecurityRule.IS_AUTHENTICATED)
    @View("securitydecorator")
    @Get
    fun index() = mapOf("books" to listOf("Developing Microservices"))
}
//end::class[]

File: test-suite-python/src/test/python/micronaut/views/model/security/BooksController.py
from micronaut.context.annotation import Requires
from micronaut.http.annotation import Controller, Get
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule
from micronaut.views import View


@Requires(property="spec.name", value="SecurityViewModelProcessorSpec")
# tag::class[]
@Controller("/")
class BooksController:

    @Secured(SecurityRule.IS_AUTHENTICATED)
    @View("securitydecorator")
    @Get
    def index(self) -> dict[str, object]:
        return {"books": ["Developing Microservices"]}
# end::class[]
